world/wguild_mgr.cpp

```diff
@@ -133,15 +141,341 @@ void WorldGuildManager::ProcessZonePacket(ServerPacket *pack) {
+	memset(&tributeData.tiers, 0, sizeof(tributeData.tiers));
+	tributeData.tier_count = 0;
+
+	tribute_list.clear();
+
+	const std::string query = "SELECT id, name, descr, unknown, isguild FROM tributes";
+	auto results = m_db->QueryDatabase(query);
+	if (!results.Success()) {
+		return false;
+	}
+
+	for (auto row = results.begin(); row != results.end(); ++row) {
+		uint32 id = Strings::ToUnsignedInt(row[0]);
+		tributeData.name = row[1];
+		tributeData.description = row[2];
+		tributeData.unknown = strtoul(row[3], nullptr, 10);
+		tributeData.is_guild = atol(row[4]) == 0 ? false : true;
+
+		tribute_list[id] = tributeData;
+	}
+
+	LogInfo("Loaded [{}] tributes", Strings::Commify(results.RowCount()));
+
+	const std::string query2 = "SELECT tribute_id, level, cost, item_id FROM tribute_levels ORDER BY tribute_id, level";
+	results = m_db->QueryDatabase(query2);
+	if (!results.Success()) {
+		return false;
+	}
+
+	for (auto row = results.begin(); row != results.end(); ++row) {
+		uint32 id = Strings::ToUnsignedInt(row[0]);
+
+		if (tribute_list.count(id) != 1) {
+			LogError("Error in LoadTributes: unknown tribute [{}] in tribute_levels", (unsigned long)id);
+			continue;
+		}
+
+		TributeData& cur = tribute_list[id];
+
+		if (cur.tier_count >= MAX_TRIBUTE_TIERS) {
+			LogError("Error in LoadTributes: on tribute [{}]: more tiers defined than permitted", (unsigned long)id);
+			continue;
+		}
+
+		TributeLevel_Struct& s = cur.tiers[cur.tier_count];
+
+		s.level = Strings::ToUnsignedInt(row[1]);
+		s.cost = Strings::ToUnsignedInt(row[2]);
+		s.tribute_item_id = Strings::ToUnsignedInt(row[3]);
+		cur.tier_count++;
+	}
+
+	LogInfo("Loaded [{}] tribute levels", Strings::Commify(results.RowCount()));
+
+	return true;
+}
+
+bool WorldGuildManager::RefreshGuild(uint32 guild_id)
+{
+	auto temp_guild = GetGuildByGuildID(guild_id);
+	BaseGuildManager::GuildInfo temp_guild_detail;
+
+	if (temp_guild) {
+		temp_guild_detail.tribute = temp_guild->tribute;
+	}
+
+	if (guild_id <= 0) {
+		LogError("Requested to refresh guild id [{}] but id must be greater than 0.", guild_id);
+		return false;
+	}
+
+	auto db_guild = BaseGuildsRepository::FindOne(*m_db, guild_id);
+	if (!db_guild.id) {
+		LogGuilds("Guild ID [{}] not found in database.", db_guild.id);
+		return false;
+	}
+
+	LogGuilds("Found guild id [{}].  Loading details.....", db_guild.id);
+	_CreateGuild(db_guild.id, db_guild.name.c_str(), db_guild.leader, db_guild.minstatus, db_guild.motd.c_str(), db_guild.motd_setter.c_str(), db_guild.channel.c_str(), db_guild.url.c_str(), db_guild.favor);
+	auto guild = GetGuildByGuildID(guild_id);
+	auto where_filter = fmt::format("guild_id = '{}'", guild_id);
+	auto g_ranks = BaseGuildRanksRepository::GetWhere(*m_db, where_filter);
+	for (auto const& r : g_ranks) {
+		guild->rank_names[r.rank] = r.title;
+	}
+
+	where_filter = fmt::format("guild_id = '{}'", guild_id);
+	auto g_permissions = BaseGuildPermissionsRepository::GetWhere(*m_db, where_filter);
```

Comment:
`g_` implies global I'd just use `permissions`

Reply:
I used g_ to represent guild_tributes to not be confused with tributes, which is used for player tributes.

Reply:
It's well understood across cpp projects that `g_` prefixed variables are global scoped. I'd prefer we remove them altogether and find another way to identify member variables such as making them private and prefixing them with `m_` 

https://stackoverflow.com/questions/22736150/global-variables-what-are-they-exactly

Reply:
Sounds good.

Reply:
Updated to be guild_ in the next commit.